js/adapt-articleView.js

```diff
@@ -1,508 +1,497 @@
+      clearTimeout(handle);
+      handle = setTimeout(callback, timeout);
+    };
+    debounced.cancel = function cancelDebounce() {
+      clearTimeout(handle);
+    };
+    return debounced;
 
-        },
+  }
 
-        _blockSliderRender: function() {
-            Adapt.trigger(this.constructor.type + 'View:preRender', this);
+  var BlockSliderView = {
 
-            this._blockSliderConfigureVariables();
+    _isReady: false,
+    _disableAnimationOnce: false,
 
-            var data = this.model.toJSON();
-            var template = Handlebars.templates['articleBlockSlider-article'];
-            this.$el.html(template(data));
+    events: {
+      "click [data-block-slider]": "_onBlockSliderClick"
+    },
 
-            this.addChildren();
+    preRender: function() {
+      AdaptArticleView.prototype.preRender.call(this);
 
-            this.$el.addClass('article-block-slider-enabled');
+      if (!this.model.isBlockSliderEnabled()) {
+        this.$el.addClass('is-disabled');
+        return;
+      }
 
-            this.delegateEvents();
+      this._blockSliderPreRender();
+    },
 
-            this.$el.imageready(_.bind(function() {
-                _.delay(_.bind(this._blockSliderPostRender, this), 500);
-            }, this));
+    _blockSliderPreRender: function() {
+      Adapt.wait.for(function(done){
+        this.resolveQueue = done;
+      }.bind(this));
+      this._blockSliderSetupEventListeners();
+    },
 
-            return this;
-        },
+    _blockSliderSetupEventListeners: function() {
 
-        _blockSliderConfigureVariables: function() {
-            var blocks = this.model.getChildren().models;
-            var totalBlocks = blocks.length;
+      this._blockSliderResizeHeight = this._blockSliderResizeHeight.bind(this);
 
-            this.model.set("_currentBlock", 0);
-            this.model.set("_totalBlocks", totalBlocks);
+      this.listenTo(Adapt, {
+        "device:resize": this._onBlockSliderResize,
+        "device:changed": this._onBlockSliderDeviceChanged,
+        "page:scrollTo": this._onBlockSliderPageScrollTo,
+        "page:scrolledTo": this._onBlockSliderPageScrolledTo
+      });
 
-            var itemButtons = [];
+      this.listenToOnce(Adapt, "remove", this._onBlockSliderRemove);
+      this.listenToOnce(this.model, "change:_isReady", this._onBlockSliderReady);
 
-            for (var i = 0, l = totalBlocks; i < l; i++) {
-                itemButtons.push({
-                    _className: (i === 0 ? "home" : "not-home") + (" i"+i),
-                    _index: i,
-                    _includeNumber: i !== 0,
-                    _title: blocks[i].get('title')
-                });
-            }
+      var duration = this.model.get("_articleBlockSlider")._slideAnimationDuration || 200;
 
-            this.model.set("_itemButtons", itemButtons);
-        },
+      this._blockSliderHideOthers = debounce(this._blockSliderHideOthers.bind(this), duration);
```

Comment:
```suggestion
      this._blockSliderHideOthers = _.debounce(this._blockSliderHideOthers.bind(this), duration);
```

And remove `debounce` stuff from top of the file